docker/synapse.Dockerfile

```diff
@@ -21,13 +21,9 @@ RUN mkdir /rust /cargo
 
 RUN curl -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain stable --profile minimal
 
-# Use the latest version of pip. This pulls in fixes not present in the
-# pip version provided by Debian Buster. See
-# https://github.com/pypa/setuptools/issues/3457#issuecomment-1190125849
 # For now, we need to tell Debian we don't care that we're editing the system python
 # installation.
 # Some context in https://github.com/pypa/pip/issues/11381#issuecomment-1399263627
-RUN ${PYTHON_VERSION} -m pip install -q --upgrade pip ${SYSTEM_PIP_INSTALL_SUFFIX}
```

Comment:
My thinking on dropping this my reading of the issue mentioned indicates that the version of pip shipped in bullseye (20.3.4-4) should be sufficient as it is greater than 19?

Reply:
The change seems sane to me (even though the sytest docker machinery does not).

I don't follow what is meant by

> For now, we need to tell Debian we don't care that we're editing the system python installation. Some context in https://github.com/pypa/pip/issues/11381#issuecomment-1399263627

which is amusing because I wrote the comment. https://github.com/matrix-org/sytest/pull/1334 and https://github.com/matrix-org/sytest/pull/1339 shed some light.

I think it's referring to `SYSTEM_PIP_INSTALL_SUFFIX`, and I think we probably need to leave this in place for now (unless every distribution is happy with the `--break-system-packages` option).

Reply:
Just to be clear, when you refer to "I think we probably need to leave this in place for now", do you mean the `SYSTEM_PIP_INSTALL_SUFFIX` on line 31/27, or are you referring to the removal on line 30? (My assumption is that you mean the former but just wanted to be sure)

Reply:
>  do you mean the SYSTEM_PIP_INSTALL_SUFFIX on line 31/27, or are you referring to the removal on line 30?

Sorry: the former, keeping SYSTEM_PIP_INSTALL_SUFFIX as the diff is currently.

Reply:
Ok great thanks!